### PR TITLE
ci: update release workflow on release branch

### PR DESCRIPTION
**Pull request by kaushik-rishi**
---
title: "Update the release workflow on the release branch (next-major-spec)"
---

---

**Related issue(s):**
Fixes https://github.com/asyncapi/spec/issues/938

---

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -11,13 +11,14 @@ jobs:
...
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
       - name: Setup Node.js
-        uses: actions/setup-node@v1
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18
       - name: Add plugin for conventional commits
-        run: npm install conventional-changelog-conventionalcommits
+        run: npm install --save-dev conventional-changelog-conventionalcommits@5.0.0
```
Comment (derberg) [inline .github/workflows/release.yml:21]:
```suggestion
        run: npm install --save-dev conventional-changelog-conventionalcommits@6.1.0
```

they release so fast 😄

**Review by derberg — COMMENTED** [review]
I'm not 100% sure all these updates will fix the release looking at the error:

```
[9:11:12 PM] [semantic-release] › ℹ  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
(node:1734) UnhandledPromiseRejectionWarning: TypeError: config.types.flatMap is not a function
    at getWriterOpts (/home/runner/work/spec/spec/node_modules/conventional-changelog-conventionalcommits/writer-opts.js:76:41)
    at module.exports (/home/runner/work/spec/spec/node_modules/conventional-changelog-conventionalcommits/writer-opts.js:47:22)
(node:1734) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:1734) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

so better if you also `npx semantic-release@17.4.3` -> `npx semantic-release@21.0.7`

I checked and the only breaking changes are npm related (not problem here) and requirement for node 18, which is already done in this PR

so we should be good to go

**Comment by derberg** [comment]
@kaushik-rishi I cannot make code change suggestion for `npx semantic-release@21.0.7` so you need to change it on your own

**Comment by kaushik-rishi** [comment]
Updated @derberg .

**Review by derberg — APPROVED** [review]
Thanks!

**Pull request merged**